File: fix_rules.py
import re

with open('firestore.rules', 'r') as f:
    content = f.read()

rules_to_add = """
    match /audit_logs/{logId} {
      allow read: if isAdmin();
      allow create: if isAuthenticated() && (isAdmin() || request.resource.data.action in ['LOGIN', 'LOGOUT']);
      allow update, delete: if false; // immutable
    }
    
    match /roles/{roleId} {
      allow read: if isAuthenticated();
      allow write: if isAdmin();
    }
"""

content = content.replace("match /settings/{settingId} {", rules_to_add + "\n    match /settings/{settingId} {")

with open('firestore.rules', 'w') as f:
    f.write(content)

File: fix_rules.js
const fs = require('fs');
let file = fs.readFileSync('firestore.rules', 'utf8');

const chaptersRule = `
    match /chapters/{chapterId} {
      allow read: if true;
      allow create: if isAuthenticated() && (isAdmin() || isOwner(get(/databases/$(database)/documents/courses/$(request.resource.data.courseId)).data.instructorId));
      allow update, delete: if isAuthenticated() && (isAdmin() || isOwner(get(/databases/$(database)/documents/courses/$(resource.data.courseId)).data.instructorId));
    }
`;

const lessonsRule = `
    match /lessons/{lessonId} {
      allow read: if true;
      allow create: if isAuthenticated() && (isAdmin() || isOwner(get(/databases/$(database)/documents/courses/$(request.resource.data.courseId)).data.instructorId));
      allow update, delete: if isAuthenticated() && (isAdmin() || isOwner(get(/databases/$(database)/documents/courses/$(resource.data.courseId)).data.instructorId));
    }
`;

file = file.replace(/match \/chapters\/\{chapterId\} \{[\s\S]*?allow write:[\s\S]*?\}/, chaptersRule.trim());
file = file.replace(/match \/lessons\/\{lessonId\} \{[\s\S]*?allow write:[\s\S]*?\}/, lessonsRule.trim());

fs.writeFileSync('firestore.rules', file);
